common/app/com/commercetools/sunrise/common/injection/SunriseModule.java

```diff
@@ -1,19 +1,50 @@
 package com.commercetools.sunrise.common.injection;
 
+import com.commercetools.sunrise.common.contexts.RequestContext;
+import com.commercetools.sunrise.common.contexts.RequestContextProvider;
+import com.commercetools.sunrise.common.contexts.RequestScope;
+import com.commercetools.sunrise.common.contexts.RequestScoped;
 import com.commercetools.sunrise.common.pages.RoutesMultiControllerComponentResolverProvider;
 import com.commercetools.sunrise.framework.MultiControllerComponentResolver;
+import com.commercetools.sunrise.hooks.RequestHookContext;
+import com.commercetools.sunrise.hooks.RequestHookContextImpl;
 import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
 import com.google.inject.name.Names;
+import io.sphere.sdk.utils.MoneyImpl;
+import play.mvc.Http;
 
 import javax.inject.Singleton;
+import javax.money.Monetary;
+import javax.money.format.MonetaryFormats;
 
 public class SunriseModule extends AbstractModule {
 
     @Override
     protected void configure() {
+        applyJavaMoneyHack();
+        bindScope(RequestScoped.class, new RequestScope());
+        bind(RequestContext.class).toProvider(RequestContextProvider.class).in(RequestScoped.class);
+        bind(RequestHookContext.class).to(RequestHookContextImpl.class).in(RequestScoped.class);
         bind(MultiControllerComponentResolver.class)
                 .annotatedWith(Names.named("controllers"))
                 .toProvider(RoutesMultiControllerComponentResolverProvider.class)
                 .in(Singleton.class);
     }
+
+    @Provides
+    @RequestScoped
```

Comment:
Cons: Duplicated code with the test
Pros: One class less and it kind of calls this only once, yet keeps the request scope.
Do you have any clue why this is never called again? But it behaves differently than when I use `@Singleton`, besides everything works perfectly.